subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -1125,6 +1150,42 @@ int bt_mesh_cfg_app_key_add(uint16_t net_idx, uint16_t addr, uint16_t key_net_id
 	return cli_wait();
 }
 
+int bt_mesh_cfg_node_reset(uint16_t net_idx, uint16_t addr, bool *status)
+{
+	BT_MESH_MODEL_BUF_DEFINE(msg, OP_NODE_RESET, 2);
+	struct bt_mesh_msg_ctx ctx = {
+		.net_idx = net_idx,
+		.app_idx = BT_MESH_KEY_DEV_REMOTE,
+		.addr = addr,
+		.send_ttl = BT_MESH_TTL_DEFAULT,
+	};
+	struct node_reset_param param = {
+		.status = status,
+	};
+	int err;
+
+	err = cli_prepare(&param, OP_NODE_RESET_STATUS);
+	if (err) {
+		return err;
+	}
+
+	bt_mesh_model_msg_init(&msg, OP_NODE_RESET);
+
+	err = bt_mesh_model_send(cli->model, &ctx, &msg, NULL, NULL);
+	if (err) {
+		BT_ERR("model_send() failed (err %d)", err);
+		cli_reset();
+		return err;
+	}
+
+	if (!status) {
+		cli_reset();
+		return 0;
+	}
+
+	return cli_wait();
```

Comment:
As the status now only depends on whether the response message was received or not, it'll either be `true` or undefined, as we never explicitly set it to `false`. It should either be set to false before sending the message, or removed.